### PR TITLE
Construct providers using DI

### PR DESCRIPTION
**Pull request by ReubenBond**
This ought to fix functional tests, which I recently broke (pretty bad at git).

Providers which are loaded from external libraries should be given access to the IoC container so that they  can access other services without resorting to globals

### CODE REVIEW

File: vNext/test/ServiceBus.Tests/ServiceBus.Tests.csproj [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="FluentAssertions" Version="4.0.0" />
+    <PackageReference Include="FluentAssertions" Version="4.18.0" />
```
Comment (jdom) [inline vNext/test/ServiceBus.Tests/ServiceBus.Tests.csproj]: I noticed this dependency once... are we even using it? If it's 1 codefile that uses it, I would prefer to just get rid of it.
I liked and used FluentAssertions, but I prefer consistency... Not necessarily to be done in this PR

Comment (ReubenBond) [inline vNext/test/ServiceBus.Tests/ServiceBus.Tests.csproj]: I'm not sure, but I saw relevant errors in vNext tests. I made this change & it fixed them

**Comment by ReubenBond** [comment]
There are some errors in newly added EventSourcingTests, but those have been failing consistently for a while.

The other failures are outputting "No gateways available" & I'm not sure what the issue there is.

**Comment by jdom** [comment]
Question is: can I merge this or are you investigating the gateway errors (or at least running those tests locally) to rule out new issues?

**Comment by xiazen** [comment]
Those No Gateway available failures can be fixed by using TestClusterPerTest in Grain_AzureBlobStore tests. I believe it is caused by Grain_AzureBlobStore tests interfered with each other... 

Basically if Grain_AzureBlobStore_SiloRestart failed, then other tests will have no silo to connect to. 

But I haven't look into why Grain_AzureBlobStore_SiloRestart would fail, it doesn't fail if run independently

**Comment by xiazen** [comment]
But I don't think it is related to Reuben's change

**Comment by ReubenBond** [comment]
Oh, thanks @xiazen! Should I make the `TestClusterPerTest` change here?

**Comment by jdom** [comment]
nah, just rule out if the change is a standard non-deterministic failure, or due to your change... Did you try running those tests locally to see if they pass? If they do, then we can fix the non-deterministic tests separately

**Comment by xiazen** [comment]
I made the TestClusterPerTest  change in my other PR . But my PR is not in a good state to review yet. you can go ahead to fix it if you want :)

**Comment by ReubenBond** [comment]
I've added the change here. A VS 2017 update yesterday (I think) destroyed my local Storage Emulator (it's gone?) ... so I need to work out how to fix that.
image

**Comment by xiazen** [comment]
? just use the real azure storage? I usually use our TestSecrects.json which is used in our load tests... just copy that file to your GitHub repo from load test VSO..

**Comment by jdom** [comment]
TBH, I don't know if moving to a slower testing approach is the way to go... Now it creates a new cluster for each test, which takes CONSIDERABLY longer than before... that's why I suggested not to fix it in this PR, and keep it isolated... if the failures were not related to your change, then I would separate the fix.

**Comment by ReubenBond** [comment]
I can ditch the commit and we can look at it in another PR - either way is fine with me. I agree that the slowness is an issue. We should address that for other tests, too.

**Comment by jdom** [comment]
Yes, please remove the commit and I'll run those tests locally. If they pass, I'll merge

**Comment by ReubenBond** [comment]
done

**Comment by jdom** [comment]
They indeed pass (as was the most likely scenario). Merged

**Pull request merged**